lib_context: update to use Irmin 3.0

craigfe opened the pull request:
Currently using the minimal indexing strategy.

Depends on https://github.com/mirage/irmin/pull/1664 (`mirage/craigfe@user-defined-indexing-strategies`), which needs the following additional pin-depends:

```
pin-depends: [
  [ "repr.dev" "git+https://github.com/mirage/repr#031cbb3728c3394e46b25a5e77aa5ceaa674c235" ]
  [ "ppx_repr.dev" "git+https://github.com/mirage/repr#031cbb3728c3394e46b25a5e77aa5ceaa674c235" ]
  [ "index.dev" "git+https://github.com/mirage/index#a60af0e40052093d810dfaca8ce5c8c1a74d882f" ]
]
```

mbouaziz committed:
Proto/Michelson: introduce parse_parameter_ty_and_entrypoints

Currently it is only `parse_parameter_ty` followed by `well_formed_entrypoints`.

This function should be used to parse the `parameter` node of a script.

mbouaziz committed:
Proto/Michelson: introduce unparse_parameter_ty

which is just `unparse_ty` with `add_field_annot`.

This function should be used to unparse the `parameter` node of a script.

mbouaziz committed:
Proto/Michelson: use parse_parameter_ty_and_entrypoints everywhere

mbouaziz committed:
Proto/Michelson: move entrypoints out of types

The pair (root_name, parameter_ty_with_annotations) is replaced by
the pair (entrypoints, parameter_ty_with_no_annotations).

And ignore union type non-entrypoint field annotations.

mbouaziz committed:
Proto/Michelson: inline add_field_annot in add_entrypoint_annot

Because add_field_annot will disappear soon

mbouaziz committed:
Proto/Michelson: return reachable flag rather than recomputing it

In `well_formed_entrypoints` and `list_entrypoints`, the reachable
flag is recomputed from the annotation, matching it the same way
as is done in the `merge` auxiliary function.

I think returning it makes the code easier to understand

mbouaziz committed:
Proto/Plugin: remove field_annot from Union type

mbouaziz committed:
Proto/Michelson: remove field_annot from Union type

mbouaziz committed:
Proto/Benchmarks: remove field_annot from Union type

mbouaziz committed:
Proto/Michelson: simplify parse_constr_annot

- Returned values were not used
- Renamed to `check_constr_annot`

mbouaziz committed:
Proto/Michelson: merge_field_annot None None is None

merge_field_annot is now dead

mbouaziz committed:
Proto/Michelson: replace extract_field_annot with remove_field_annot

which is the same thing without the second return value

and with `has_field_annot` which is `extract_field_annot` composed with
`Option.is_some`

mbouaziz committed:
Proto/Michelson: introduce parse_entrypoint_annot_strict

which is exactly `parse_entrypoint_annot composed` with
`field_annot_opt_to_entrypoint_strict`

mbouaziz committed:
Proto/Michelson: move parse_passable/any_ty later

So that `parse_ty` is the first one in the mutually recursive definition

mbouaziz committed:
Proto/Michelson: move parse_packable_ty later

It doesn't need to be in the mutually recursive definition

mbouaziz committed:
Proto/Michelson: move parse_view_input/output_ty later

They don't need to be in the mutually recursive definition

mbouaziz committed:
Proto/Michelson: move parse_normal_storage_ty later

It doesn't need to be in the mutually recursive definition

mbouaziz committed:
Proto/Michelson: remove Inconsistent_(type_)annotations errors

abate committed:
ci: run tezt tests for static binaries

abate committed:
test: add fast python tests for static binaries

abate committed:
shellcheck: upload-static-binaries-to-package-registry.sh and release/binaries.sh

icristescu committed:
lib_context: update to use Irmin 3.0

... and avoid passing an index to post-snapshot commit validation.

Co-authored-by: Craig Ferguson <me@craigfe.io>

icristescu committed:
lib_context: use the minimal indexing strategy

Commit:
tezt_long_test: Generalize function [time] and [time_lwt]

abate committed:
ci: publish tarball on gitlab registry

killian-delarue committed:
lib_openapi: Add Method module

Commit:
Changes: add entry

raphael-proust committed:
Client: convert code from infix bind into let binding operators

raphael-proust committed:
Validation: convert code from infix bind into let binding operators

craigfe committed:
lib_context: always index during snapshot import

This is necessary in order to resolve internal pointers the children of
a parent object when importing it (as the snapshot format references
children by hash).

raphael-proust committed:
Client: minor stylistic improvements

raphael-proust committed:
Validation: minor code cosmetics

novalis committed:
Tickets: refactor Ticket_hash_repr into two modules

So that we can use the Ticket_hash_repr above the context

ACoquereau committed:
Protos/Plugin: add generators for filter_state

ACoquereau committed:
Protos/Plugin: Add test_utils file for filter_state

ACoquereau committed:
Protos/Plugin: Add tests for filter_state manager functions

ACoquereau committed:
Protos/Plugin: Add flush test

ferilata committed:
Proto: SCORU: Add commitment representation and storage

Commit:
ocaml-ledger: add support to tenderbake changes

Commit:
Client/Ledger: add rounds support on high watermark requests/resets

Ngoguey42 commented:
I've rebased on top of a newer master. It uses repr 0.9.0, the same index version and irmin main at hash `b5b8186ca9eeafb1fb5ea34fea3417d7e8931457`.

mbouaziz committed:
Proto/Tests: bump lambda size expected mean

I got a little bit more than 1.28 :-/

See #2327 for how to fix this

ACoquereau committed:
Lib_test: Add qcheck_cond to qcheck2_helpers

Commit:
tezt_perf_reg: Init package

Commit:
tezt_long_test: move long_tests library files into tezt-perf-reg

Commit:
tezt: Updates and uniformize package dependencies

This commit removes the dependency from [lib_performance_regression]
to [lib_tezos], as the protocol seems out of scope.

It inverts this dependency, moving the function
[Long_test.register_with_protocol] inside [Protocol] module.

Commit:
tps_evaluation: add tezt-performance-regression dependency

Commit:
tps_evaluation: Enables perf. reg. check feature

Commit:
tps_evaluation: update makefile to ensure tezt is well compiled

lthms committed:
Proto,tx_rollup: Add a L1 operation to fill the inbox of L2 batches

Co-authored-by: Marigold <contact@marigold.dev>
Co-authored-by: Nomadic Labs <contact@nomadic-labs.com>
Co-authored-by: Oxhead Alpha <info@oxheadalpha.com>

lthms committed:
Client,tx_rollup: Add a command to submit a batch of L2 operations

Co-authored-by: Marigold <contact@marigold.dev>
Co-authored-by: Nomadic Labs <contact@nomadic-labs.com>
Co-authored-by: Oxhead Alpha <info@oxheadalpha.com>

lthms committed:
Proto,tx_rollup: Unit tests for submitting batches of L2 operations

Co-authored-by: Marigold <contact@marigold.dev>
Co-authored-by: Nomadic Labs <contact@nomadic-labs.com>
Co-authored-by: Oxhead Alpha <info@oxheadalpha.com>

lthms committed:
Tezt,tx_rollup: Provide helpers to submit batches of L2 operations

Co-authored-by: Marigold <contact@marigold.dev>
Co-authored-by: Nomadic Labs <contact@nomadic-labs.com>
Co-authored-by: Oxhead Alpha <info@oxheadalpha.com>

lthms committed:
Tezt,tx_rollup: Add an integration test that submits a batch

Co-authored-by: Marigold <contact@marigold.dev>
Co-authored-by: Nomadic Labs <contact@nomadic-labs.com>
Co-authored-by: Oxhead Alpha <info@oxheadalpha.com>

Commit:
tps_evaluation: Use logging module to log

Commit:
tezt_long_test: Update long tests documentation

killian-delarue committed:
lib_openapi: add of_json to convert json to openapi type

iguerNL committed:
Shell/Doc: pending operations ordering and prioritization

Commit:
tps_evaluation: Documentation

mbouaziz committed:
Proto/Michelson: turn root_name into an Entrypoint.t

That way we'll get errors earlier.

picdc-work committed:
Client/Proto_context: expose transfer version with expr as parameter

Commit:
Manifest: add open_

mbouaziz committed:
Proto/Michelson: introduce parse_entrypoint_annot_lax

zshipko committed:
bin_node: storage head-commit command to print current head hash

iguerNL committed:
Plugin/Doc: pending operations ordering, prioritization and limits

Co-Author: Nic Volanschi <nic.volanschi@nomadic-labs.com>

NicNomadic committed:
doc: move .gitignore docs/* items to docs/.gitignore

raphael-proust committed:
Validation: simplify control-flow

iguerNL committed:
Shell/Prevalidator: count operations precheck successes (under-approx)

iguerNL committed:
Proto/Plugin: Check signature of previously prechecked operations once

mbouaziz committed:
Proto/Doc: add entry in changelog

mbouaziz committed:
Proto/Michelson: update ty_size

mbouaziz committed:
Proto/Tests: adapt script cache hard-coded value

raphael-proust committed:
Lwtreslib: add latest Stdlib functions to List

raphael-proust committed:
Lwtreslib: consmetics: more uniform code in tests

raphael-proust committed:
Lwtreslib: add newest Stdlib functions to Map

raphael-proust committed:
Lwtreslib: add latest Stdlib functions to Set

raphael-proust committed:
Lwtreslib: add latest Stdlib functions to Seq

raphael-proust committed:
Lwtreslib: Update Hashtbl to latest Stdlib

Commit:
Signer: always take into account '--password-filename' when trying to 'decrypt' a key

raphael-proust committed:
Lwtreslib: document implementation detail about recursive functions

novalis committed:
Tezt: elide operation hashes

iguerNL committed:
Tests/Tezt: ill-signed op in branch_delayed reclassified as refused

Sventimir committed:
Proto Alpha: Make Gas.Arith.t type private.

Commit:
Signer: take into account password_filename (and use the same long name as tezos-client)

vch9 committed:
Tezt,tx_rollup: Test invalid cases of submit

mbouaziz committed:
Proto/Michelson: simplify unparse_ty_uncarbonated

`annot` was always `[]`

mbouaziz committed:
Proto/Michelson: rewrite unparse_ty_uncarbonated

so as to makes it clearer the result is always a `Prim`

Also it will make it easier to add an annotation in case
of entrypoints, avoiding the imperfect `add_field_annot`.

mbouaziz committed:
Proto/Michelson: remove assert false for unparse_parameter_ty

eugenz-nl committed:
Docs/User: fix referred value of preserved_cycles

The doc was using the value from the sandbox mode instead of the mainnet value.

mbouaziz committed:
Proto/Michelson: rewrite parse_ty

to ease a later diff

mbouaziz committed:
Proto/Michelson: integrate parsing of entrypoints into parse_ty

mbouaziz committed:
Proto/Michelson: keep root name in parameter type annotation

mbouaziz committed:
Proto/Michelson: rename parse_parameter_ty into parse_passable_ty

This is to make it clearer the difference between parsing a type
that can be a passed as a parameter and parsing the type of the
`parameter` node of a script.

Note that passable is the terminology used in the Michelson
reference documentation.

Commit:
Context: lower the default lru allocated size

eugenz-nl committed:
Shell_service: use right constant in docstring

eugenz-nl committed:
lib_env: add clarification in docstring

mbouaziz committed:
Proto/Michelson: introduce extract_entrypoint_annot

which is just `extract_field_annot` composed with
`Entrypoint.of_annot_lax_opt`

Commit:
CI: Lock docker version to 20.10.12 everywhere

ferilata committed:
Env: Add Bounded from lib_base to Environment

Also add it to Tezos_base.TzPervasives

Commit:
CI: Use '.image_template__latest' template everywhere and refactor

Commit:
Normalize docker job names

Commit:
Remove unused/useless tag

Commit:
Allow to interrupt and/or retry Docker jobs, it is not a problem

Commit:
doc: add links for InfluxDB and Grafana

pays a small comms service to OS tools we rely on.

mbouaziz committed:
Proto/Michelson: small rewrites of parse_ty

to reduce next diff size

mbouaziz committed:
Proto/Michelson: introduce parse_ty_ret

to make next diff smaller

Commit:
Only manual jobs should not be interruptible

mbouaziz committed:
Tezt: update regression tests outputs

mbouaziz committed:
Tests/Python: update typechecking traces

mbouaziz committed:
Proto/Michelson: update kinstr_size

to count entrypoints too

Commit:
Opam: update opam dependencies

vect0r-vicall committed:
Context: clearer default values and documentation for constants

vect0r-vicall committed:
Doc/storage: describe available environment variables

raphael-proust committed:
Manifest: support `(documentaiton)` stanza

raphael-proust committed:
Docs: instruct documenters how to add documentation stanza

raphael-proust committed:
Manifest: add `~documentation` to appropriate packages

vect0r-vicall committed:
Doc: crossref TEZOS_CONTEXT environment variable

killian-delarue committed:
Doc: OpenAPI for Ithaca (using v12.0~rc2)

killian-delarue committed:
Doc: update OpenAPI doc generator to Ithaca

mebsout committed:
Doc/CHANGES: move RPC change in 12.0~rc2 in correct section

mbouaziz committed:
Tests/Python: update gas in regtests outputs

ACoquereau committed:
Tezt: enable and fix mempool tests for ithaca

Commit:
tps-evaluation: Use PRT framework for the gas-tps measurement

Commit:
Opam: manually remove 'hidden' packages

Commit:
tezt_long_test: Update [time] related functions to enable tagging

Commit:
Proto/Plugin: modify plugin description

saroupille committed:
Protocol: Explicit default case for precheck

saroupille committed:
Protocol: Do not propagate operations with feature flag

novalis committed:
Proto,tx_rollup: Store a linked list of inbox levels

Co-authored-by: Marigold <contact@marigold.dev>
Co-authored-by: Nomadic Labs <contact@nomadic-labs.com>
Co-authored-by: Oxhead Alpha <info@oxheadalpha.com>

raphael-proust committed:
Proto-env: adapt to updates in Lwtreslib

- Update env-v5 along with Lwtreslib
- Restrict env-v3 and env-v4 sets and maps so that their `S` component
  is backwards compatible.

mrkkrp committed:
tps_evaluation: fix the function for weighted average

This should resolve the problem with 0 empirical TPS when average block is
specified explicitely.

Commit:
Merge branch 'openapi-ithaca2' into 'master'

Doc: OpenAPI for Ithaca (using v12.0~rc2)

See merge request tezos/tezos!4325

Commit:
Merge branch 'francois@oru-feature-flag' into 'master'

Protocol: Do not propagate operations conditioned by a feature flag

See merge request tezos/tezos!4330

Commit:
Merge branch 'raphael-p@documentation-stanza-for-libraries-with-mlds' into 'master'

Documentation stanza for libraries with mlds

See merge request tezos/tezos!4324

Commit:
Docs: enable parallel processing

Commit:
Merge branch 'abate@tarballs' into 'master'

Create tarballs for tezos releases.

Closes #1751

See merge request tezos/tezos!3635

zshipko committed:
Changelog: add tezos-node storage head-commit

Commit:
Merge branch 'hans@scoru-commit-storage-wip' into 'master'

SCORU: Add storage of commitments

Closes #2118 and #2107

See merge request tezos/tezos!4148

Commit:
Merge branch 'tx_rollup@batches' into 'master'

Tx_rollup: Add the [Tx_rollup_submit_batch] operation

See merge request tezos/tezos!4203

Commit:
Merge branch 'vbot@add_round_support_ledger' into 'master'

Add rounds support on high watermark requests/resets to ledger

See merge request tezos/tezos!4047

let-def committed:
Packer: add indirections to split initialisation of sub-modules to sub-functions

Directly concatenating all modules produce a unique, big, control-flow-graph for
the whole initialisation.
Wrapping the sub-modules in functors that are immediately applied split
the initialisation procedure in smaller components.

Auditing the CMM shows that this wrapping does not prevent
optimizations/inlining.

Commit:
Merge branch 'storage-head-command' into 'master'

bin_node: Add `storage head-commit` command to print current context head hash

See merge request tezos/tezos!4285

raphael-proust committed:
Scripts/yes-wallet: convert code from infix bind into let binding operators

Commit:
Merge branch 'raphael-p@convert-src-bin-validation-to-newer-error-monad-syntax' into 'master'

Convert src/bin_validation to newer error monad syntax

See merge request tezos/tezos!4094

mbouaziz committed:
Tests/Python: update regtests outputs

raphael-proust committed:
Client-base-unix: convert code from infix bind into let binding operators

mbouaziz committed:
Proto/Michelson: remove dead field_annot_opt_eq_entrypoint_lax

Commit:
Merge branch 'password_filename_in_decrypt' into 'master'

Password-filename and tezos-signer

Closes #2361

See merge request tezos/tezos!4299

raphael-proust committed:
Client-commands: convert code from infix bind into let binding operators

Commit:
Merge branch 'abate@static-testing' into 'master'

Add  python integration tests and tezt tests for static binaries

See merge request tezos/tezos!3322

Commit:
Opam: freese the version of json_data_encoding for now

raphael-proust committed:
Context: convert code from infix bind into let binding operators

raphael-proust committed:
Context: minor code cosmetics

raphael-proust committed:
Crypto: switch from Lwt.Infix to Lwt.Syntax

Commit:
Manifest: use open_ instead of ~opens

Commit:
Manifest: regenerate dune and .opam files

raphael-proust committed:
Micheline: convert code from infix bind into let binding operators

raphael-proust committed:
Rpc: convert code from infix bind into let binding operators

Commit:
Merge branch 'raphael-p@convert-lib-crypto-to-newer-let-syntax' into 'master'

Crypto: switch from Lwt.Infix to Lwt.Syntax

See merge request tezos/tezos!4353

Commit:
Lint: only put .ocamlformat files next to dune-project

Commit:
Merge branch 'albin@fix-ithaca-mempool-tezt-tests' into 'master'

Tezt: enable and fix mempool tests for ithaca

See merge request tezos/tezos!4327

Commit:
Update .ocamlformat files

Commit:
Merge branch 'hhugo-update' into 'master'

Everywhere: update opam deps

See merge request tezos/tezos!4322

mbouaziz committed:
Proto/Michelson: use Gas_monad for find_entrypoint

mbouaziz committed:
Proto/Michelson: push Gas_monad further in find_entrypoint

mbouaziz committed:
Proto/Michelson: carbonate find_entrypoint

Fixes #2423

The first comparison is not carbonated to avoid changing all the
traces. As it's always done once, it can be taken into account
by callers.

mbouaziz committed:
Tests/Python: update typechecking traces

mbouaziz committed:
Tests/Python: update typechecking regtests outputs

mbouaziz committed:
Proto/Michelson: remove legacy behaviour in parse_toplevel

Commit:
Merge branch 'raphael-p@convert-lib-micheline-to-newer-let-syntax' into 'master'

Micheline: convert code from infix bind into let binding operators

See merge request tezos/tezos!4354

killian-delarue committed:
lib_openapi: fix parameter enum of Schema Integer and String

mbouaziz committed:
Tests/Python: update liquidity baking regtests outputs

Commit:
Merge branch 'sventimir@abstract_gas_repr' into 'master'

Proto Alpha: Make Gas.Arith.t type private.

See merge request tezos/tezos!4293

mbouaziz committed:
Proto/Michelson: deprecate annotation on parameter toplevel constructor

mbouaziz committed:
Proto/Doc: add entry to changelog

mbouaziz committed:
Proto/Doc: do not use it in the documentation

mbouaziz committed:
Tests/Python: adapt tests 1/3

mbouaziz committed:
Tests/Python: adapt tests 3/3

mbouaziz committed:
Tests/Python: adapt tests 2/3

Commit:
Merge branch 'fix-stack-overflow' into 'master'

protocol_compiler: split the large functor into chunks

See merge request tezos/tezos!4340

Commit:
Merge branch 'mehdi@legacy_parse_toplevel' into 'master'

Proto/Michelson: remove legacy behaviour in parse_toplevel

Closes #2422

See merge request tezos/tezos!4364

NicNomadic committed:
doc: optimize tezos_custom_role.py using a cache

Commit:
Merge branch 'lthms@ticket-hash-repr-refactor' into 'master'

Proto: Break the dependency between `Ticket_hash_repr' and `Raw_context'

See merge request tezos/tezos!4323

Commit:
Merge branch 'vbot@reduce_lru_size' into 'master'

Lower Irmin's default lru size

See merge request tezos/tezos!4310

Commit:
Build: remove unused deps

Commit:
Merge branch 'raphael-p@convert-src-bin-client-to-newer-error-monad-syntax' into 'master'

Convert src/bin_client to newer error monad syntax

See merge request tezos/tezos!4090

Commit:
Opam: update the world

Commit:
Merge branch 'killian@openapi_from_json' into 'master'

add from_json function in lib_openapi/openapi.ml

Closes #2151

See merge request tezos/tezos!3918

raphael-proust committed:
Rpc, Rpc-http: avoid the Lwt. prefix when the value is in scope

Commit:
Merge branch 'romain-manifest-open' into 'master'

Manifest: introduce `open_` to replace `~opens`

See merge request tezos/tezos!4355

Commit:
Merge branch 'hhugo-rm-ocamlformat' into 'master'

Lint: reduce the number of dot ocamlformat files in the tree

See merge request tezos/tezos!4358

Commit:
Merge branch 'raphael-p@convert-lib-rpc-to-newer-let-syntax' into 'master'

Rpc: convert code from infix bind into let binding operators

See merge request tezos/tezos!4356

NicNomadic committed:
doc: disable pylint warning global-statement

Commit:
Merge branch 'raphael-p@convert-yes-wallet-to-newer-let-syntax' into 'master'

Scripts/yes-wallet: convert code from infix bind into let binding operators

See merge request tezos/tezos!4345

smelc committed:
lib_proxy: Proxy: document undocumented public symbols

smelc committed:
lib_proxy: renaming in Proxy.TREE: set_leaf->add_leaf

Commit:
Merge branch 'raphael-p@convert-lib-client-base-unix-to-newer-let-syntax' into 'master'

Client-base-unix: convert code from infix bind into let binding operators

See merge request tezos/tezos!4347

novalis committed:
Proto,tx_rollup: Pass state into append_message

Commit:
Manifest: rename Target.target into Target.t

Commit:
Manifest: make preprocessor type abstract

Commit:
Manifest: add no_target, if_some and if_

Commit:
Merge branch 'killian@openapi_enum_default_value' into 'master'

Openapi: Refactor schema type to fix enum parameter default value

See merge request tezos/tezos!4379

Commit:
Merge branch 'functori@prevalidator-check-signature-once' into 'master'

Check signature of successfully prechecked operations only once

See merge request tezos/tezos!4255

Commit:
Merge branch 'albin@test-plugin' into 'master'

Tests/Plugin: Test one manager restriction in the plugin

Closes #2238 and #2181

See merge request tezos/tezos!4016

Commit:
Merge branch 'smelc@lib-proxy-proxy-add-doc' into 'master'

lib_proxy: Proxy: document undocumented public symbols

Closes #2432

See merge request tezos/tezos!4381

mbouaziz committed:
Proto/Michelson: adapt gas for find_entrypoint

The function has been rewritten in a previous commit.

mbouaziz committed:
Tests/Python: update gas in regtests outputs

Commit:
Merge branch 'mehdi@deprecate_parameter_annot' into 'master'

Proto/Michelson: deprecate parameter annot

See merge request tezos/tezos!4366

Commit:
Manifest: fix node_wrapper_flag handling

Commit:
Manifest: keep version of JS package in the manifest

Sventimir committed:
Proto Alpha: Add `consumed_milligas` to the block metadata encoding.

This is duplicate of `consumed_gas` field introduced for uniformity
with other encodings.

Sventimir committed:
Docs: deprecated `consumed_gas` in encodings.

Encourage use of `consumed_milligas` instead. Deprecated field will
be removed form encodings in a follow-up MR (see issue #2246).

Commit:
Tezt: fix long tests which no longer compiled

mbouaziz committed:
Tezt: adapt gas in regtests outputs

Commit:
CI: type-check tezt/long_tests

Commit:
Merge branch 'hhugo-update-deps' into 'master'

Build: remove unused deps

See merge request tezos/tezos!4371

saroupille committed:
CHANGES: Deprecate the `max_refused_operations` parameter

Commit:
Merge branch 'romain-fix-long-tests' into 'master'

Tezt: fix long tests which no longer compiled

See merge request tezos/tezos!4389

Commit:
Merge branch 'raphael-p@lwtreslib-updates-from-stdlib-4-13' into 'master'

Lwtreslib updates from stdlib 4.13

See merge request tezos/tezos!4279

Commit:
Merge branch 'francois@deprecate-mempool-max-refused-operations' into 'master'

CHANGES: Deprecate the `max_refused_operations` parameter

See merge request tezos/tezos!4393

Commit:
tezt: Fix estimate gas computation

abate committed:
ci: add back curl in publish_release.yml

Commit:
Merge branch 'raphael-p@convert-lib-client-commands-to-newer-let-syntax' into 'master'

Client-commands: convert code from infix bind into let binding operators

See merge request tezos/tezos!4348

Commit:
Merge branch 'davdumas@normalize-job-names' into 'master'

CI: Normalize Docker jobs - Names, tags and interactability

See merge request tezos/tezos!4319

saroupille committed:
Protocol/Tx_rollup: Make message type safe

saroupille committed:
Protocol/Tx_rollup: Compute the size at the beginning

Commit:
Merge branch 'davdumas@template-docker' into 'master'

CI: Docker locked version with configuration template

See merge request tezos/tezos!4316

novalis committed:
Tx_rollup, proto: rearrange message size check to before inbox manipulation

Commit:
Merge branch 'arvid@docs-parallel-processing' into 'master'

Docs: Optimize the Tezos custom RST roles

See merge request tezos/tezos!4368

NicNomadic committed:
doc: delete ignored file user/default-acl.json in Makefile

Commit:
Merge branch 'hhugo-fix-node-wrapper' into 'master'

Fix node-wrapper  handling in the manifest

See merge request tezos/tezos!4386

Commit:
Merge branch 'raphael-p@convert-lib-context-to-newer-let-syntax' into 'master'

Convert lib context to newer let syntax

See merge request tezos/tezos!4352

Commit:
Merge branch 'functori@fix-missing-changes-mempool-replace-gc' into 'master'

Doc/CHANGES: move RPC change in 12.0~rc2 in correct section

See merge request tezos/tezos!4326

Commit:
Merge branch 'tx_rollup@inbox-linked' into 'master'

Proto,tx_rollup: Store a linked list of inbox levels

See merge request tezos/tezos!4332

Commit:
Merge branch 'pierrick@client-transfer-without-parsing' into 'master'

Client/Proto_context: expose transfer version with expr as parameter

See merge request tezos/tezos!4263

NicNomadic committed:
doc: replace carrot by cannot

Commit:
Merge branch 'mehdi@carbonate_find_entrypoint' into 'master'

Proto/Michelson: carbonate find_entrypoint

Closes #2423

See merge request tezos/tezos!4363

Commit:
Merge branch 'romain-test-build-long' into 'master'

CI: type-check tezt/long_tests

Closes #2438

See merge request tezos/tezos!4391

Commit:
Merge branch 'hhugo-update-2' into 'master'

Opam: update the world

See merge request tezos/tezos!4372

Commit:
Docs/clic: tutorial

Commit:
Merge branch 'arvid@clic-doc' into 'master'

Docs/clic: tutorial

See merge request tezos/tezos!1952

Commit:
Merge branch 'romain-manifest-no-target' into 'master'

Manifest: add `no_target` target

See merge request tezos/tezos!4380

Commit:
Merge branch 'mehdi@ignore_annotations_5.0' into 'master'

Proto/Michelson: move entrypoints out of types

See merge request tezos/tezos!4311

raphael-proust committed:
Test: convert from infix to binding operators

Note that these infix/bindings are from Qcheck

Commit:
Merge branch 'nray@tps-as-long-test' into 'master'

Use the modules from Tezt's "long test" to register results of TPS bechmark

Closes #2319

See merge request tezos/tezos!4207

Commit:
Merge branch 'abate@curl' into 'master'

ci: add back curl in publish_release.yml

See merge request tezos/tezos!4400

Commit:
Merge branch 'mehdi@ignore_annotations_5' into 'master'

Proto/Michelson: remove all field_annot

See merge request tezos/tezos!4259

Commit:
Merge branch 'raphael-p@convert-lib-test-to-newer-let-syntax' into 'master'

Test: convert from infix to binding operators

See merge request tezos/tezos!4419

Commit:
Merge branch 'functori@doc-prevalidator' into 'master'

Doc: Pending operations ordering, prioritization and limits

See merge request tezos/tezos!4227

Commit:
Merge branch 'sventimir@deprecate-consumed-gas' into 'master'

Proto Alpha: Deprecate `consumed_gas` field in encodings.

Closes #2246

See merge request tezos/tezos!4388

Commit:
Merge branch 'typo-doc' into 'master'

doc: typo train

See merge request tezos/tezos!4243
